cardinal/world.go

```diff
@@ -54,22 +54,27 @@ type (
 )
 
 // NewWorld creates a new World object using Redis as the storage layer.
-func NewWorld(addr, password string, opts ...WorldOption) (*World, error) {
+func NewWorld(opts ...WorldOption) (*World, error) {
 	ecsOptions, serverOptions, cardinalOptions := separateOptions(opts)
-	log.Info().Msg("Running in normal mode, using external Redis")
-	if addr == "" {
-		return nil, errors.New("redis address is required")
-	}
-	if password == "" {
-		log.Info().Msg("Redis password is not set, make sure to set up redis with password in prod")
-	}
 
+	cfg := GetWorldConfig()
+
+	if cfg.CardinalDeployMode == "production" {
+		log.Logger.Info().Msg("Starting a new Cardinal world in production mode")
+		if cfg.RedisPassword == "" {
+			return nil, errors.New("redis password is required in production")
+		}
+	} else {
+		if cfg.CardinalDeployMode != "development" {
+			log.Logger.Warn().Msg("CARDINAL_DEPLOY_MODE is unrecognized. Defaulting to development mode")
+		}
+		log.Logger.Info().Msg("Starting a new Cardinal world in development mode")
+	}
```

Comment:
not really sure what this control flow is supposed to control. Is "development" supposed to start world with miniredis?

Reply:
Dev mode just doesn't enforce redis password and I wanted to set the logger to pretty print as well but I forgot